analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/Verifier.cs

```diff
@@ -54,83 +57,180 @@ public static void VerifyNoExceptionThrown(string path, IEnumerable<DiagnosticAn
             DiagnosticVerifier.GetAllDiagnostics(compilation, diagnosticAnalyzers, checkMode);
         }
 
+        /// <summary>
+        /// Verify analyzer from C# on a snippet in non-concurrent execution mode.
+        /// </summary>
         public static void VerifyCSharpAnalyzer(string snippet,
                                         SonarDiagnosticAnalyzer diagnosticAnalyzer,
                                         IEnumerable<MetadataReference> additionalReferences) =>
             VerifyCSharpAnalyzer(snippet, diagnosticAnalyzer, null, CompilationErrorBehavior.Default, additionalReferences);
 
+        /// <summary>
+        /// Verify analyzer from C# on a snippet in non-concurrent execution mode.
+        /// </summary>
         public static void VerifyCSharpAnalyzer(string snippet,
                                         SonarDiagnosticAnalyzer diagnosticAnalyzer,
                                         CompilationErrorBehavior checkMode,
                                         IEnumerable<MetadataReference> additionalReferences = null) =>
             VerifyCSharpAnalyzer(snippet, diagnosticAnalyzer, null, checkMode, additionalReferences);
 
+        /// <summary>
+        /// Verify analyzer from C# on a snippet in non-concurrent execution mode.
+        /// </summary>
         public static void VerifyCSharpAnalyzer(string snippet,
                                                 SonarDiagnosticAnalyzer diagnosticAnalyzer,
                                                 IEnumerable<ParseOptions> options = null,
                                                 CompilationErrorBehavior checkMode = CompilationErrorBehavior.Default,
                                                 IEnumerable<MetadataReference> additionalReferences = null)
         {
             var solution = SolutionBuilder.Create().AddProject(AnalyzerLanguage.CSharp).AddSnippet(snippet).AddReferences(additionalReferences).GetSolution();
-            VerifyAnalyzer(solution, new DiagnosticAnalyzer[] { diagnosticAnalyzer }, options, checkMode);
+            CompileAndVerifyAnalyzer(solution, new DiagnosticAnalyzer[] { diagnosticAnalyzer }, options, checkMode);
         }
 
+        /// <summary>
+        /// Verify analyzer from C# on a series of snippets in non-concurrent execution mode.
+        /// </summary>
+        public static void VerifyCSharpAnalyzer(string[] snippets,
```

Comment:
This one exists only for testing another class `DiagnosticVerifier`. Let's remove it and do the minimal scaffolding in `DiagnosticVerifierTest`.This does way more things that we actually need to test that functionality.

As alternative, you can use another method and move those two snippets to a TestCases/somewhere